docs/plugin/messaging.md

```diff
@@ -0,0 +1,886 @@
+
+```c++
+Config()
+    : Core::JSON::Container()
+    , Console(false)
+    , SysLog(false)
+    , FileName()
+    , Abbreviated(true)
+    , MaxExportConnections(Publishers::WebSocketOutput::DefaultMaxConnections)
+    , Remote()
+{
+    Add(_T("console"), &Console); // (1)
+    Add(_T("syslog"), &SysLog); // (2)
+    Add(_T("filepath"), &FileName); // (3)
+    Add(_T("abbreviated"), &Abbreviated); // (4)
+    Add(_T("maxexportconnections"), &MaxExportConnections); // (5)
+    Add(_T("remote"), &Remote); // (6)
+}
+```
+
+1. Boolean value indicating if console output is enabled/disabled
+2. Boolean value indicating if syslog output is enabled/disabled
+3. Path to the file in which the messages should be stored. If the path is not empty, the file output is enabled
+4. Reducing the amount of information in the messages coming from the `MessageControl` plugin (e.g. timestamp reduced to the time of day instead of the full date; removing file name, line number and class name for tracing type messages)
+5. Specifying to how many WebSockets can the messages be outputted
+6. An object that should have two properties: `binding` which corresponds to a binding address and `port` on which the UDP connection will be established
+
+!!! note
+	Even though by default no output is set to either true or false in this config file, the plugin will output the messages to a console or syslog depending on whether Thunder is running in the background or not.
+
+## DirectOutput
+
+It is important to note that when the `MessageControl` plugin is not actively running or in scenarios where it is disabled, there is an option to directly print messages onto the console using the `DirectOutput()` method. This allows for immediate display of messages on the console without going through the buffering process.  In such a situation, the `Output()` method of the `DirectOutput` class is used. Furthermore, the `DirectOutput` class can be configured using the `Mode()` method to send the messages to the system logger instead of simply printing them on the console.
+
+```c++
+void DirectOutput::Output(const Core::Messaging::MessageInfo& messageInfo, const Core::Messaging::IEvent* message) const
+{
+    ASSERT(message != nullptr);
+    ASSERT(messageInfo.Type() != Core::Messaging::Metadata::type::INVALID);
+
+    string result = messageInfo.ToString(_abbreviate).c_str() +
+                    Core::Format("%s\n", message->Data().c_str());
+
+#ifndef __WINDOWS__
+    if (_isSyslog == true) {
+        //use longer messages for syslog
+        syslog(LOG_NOTICE, "%s\n", result.c_str());
+    }
+    else
+#endif
+    {
+        std::cout << result << std::endl;
+    }
+}
+```
+
+
+# How to adjust messaging
+
+## Configuration
+
+The config file can be used to enable/disable the default messaging categories used for logging, tracing and warning reporting.
```

Comment:
Be explicit and state that this is the Thunder config file in /etc/WPEFramework/config.json, not a plugin config file